src/routes/safe/components/Apps/confirmTransactions.tsx

```diff
@@ -41,21 +42,43 @@ const IconText = styled.div`
   }
 `
 const isTxValid = (t) => {
+  try {
+    if (!['string', 'number'].includes(typeof t.value)) {
+      return false
+    }
+
+    if (typeof t.value === 'string') {
+      const web3 = getWeb3()
+      web3.eth.abi.decodeParameter('uint256', t.value)
+    }
+  } catch (error) {
+    return false
+  }
+
   const isAddressValid = mustBeEthereumAddress(t.to) === undefined
-  return isAddressValid && t.value !== undefined && typeof t.value === 'number' && t.data && typeof t.data === 'string'
+  return isAddressValid && t.data && typeof t.data === 'string'
 }
 
 const confirmTransactions = (
-  safeAddress,
-  safeName,
-  ethBalance,
-  nameApp,
-  iconApp,
-  txs,
-  openModal,
-  closeModal,
-  onConfirm,
-) => {
+  safeAddress: string,
+  safeName: string,
+  ethBalance: string,
+  nameApp: string,
+  iconApp: string,
+  txs: Array<{
```

Comment:
can we export
```
{
  to: string
  value: string | number
  data: string
}
```
to a separate type? Let's say `SafeAppTx`

Reply:
done